skfda/misc/metrics.py

```diff
@@ -279,30 +279,50 @@ def norm_lp(fdatagrid, p=2, p2=2):
     if not (p == 'inf' or np.isinf(p)) and p < 1:
         raise ValueError(f"p must be equal or greater than 1.")
 
-    if fdatagrid.dim_codomain > 1:
-        if p2 == 'inf':
-            p2 = np.inf
-        data_matrix = np.linalg.norm(fdatagrid.data_matrix, ord=p2, axis=-1,
-                                     keepdims=True)
-    else:
-        data_matrix = np.abs(fdatagrid.data_matrix)
+    if isinstance(fdata, FDataBasis):
+        if fdata.dim_codomain > 1 or p != 2:
+            raise ValueError
 
-    if p == 'inf' or np.isinf(p):
+        res = np.empty(fdata.n_samples)
 
-        if fdatagrid.dim_domain == 1:
-            res = np.max(data_matrix[..., 0], axis=1)
-        else:
-            res = np.array([np.max(sample) for sample in data_matrix])
+        gram = fdata.basis.gram_matrix()
```

Comment:
I need a comment here. I do not know what is happening.

Reply:
Basically I compute the lower triangular matrix and the diagonal separately, then, as the Gram matrix is symmetric the final result is the diagonal plus two times the lower triangular matrix.

Reply:
I included a comment on the code explaining the procedure.